corehq/apps/locations/fixtures.py

```diff
@@ -229,57 +230,83 @@ def should_sync_flat_fixture(project, app):
-
-        user_locations = restore_user.get_sql_locations(restore_user.domain)
-        distance_dict = LocationRelation.relation_distance_dictionary(user_locations)
-
-        child_node = xml_nodes.getchildren()[0]
-        for grandchild_node in child_node.getchildren():
-            location_id = grandchild_node.get('id')
-            if location_id in distance_dict:
-                minimum_distance = min(six.itervalues(distance_dict[location_id]))
-                grandchild_node.set('distance', str(minimum_distance))
-
-        return [get_index_schema_node(fixture_id, attrs_to_index), xml_nodes]
-
-
 class RelatedLocationsFixtureProvider(FixtureProvider):
-    """This fixture is under active development for REACH, and is expected to change.
-
-    - Relations do not nest. Meaning that a location needs a direct relation
-      to another location for it be included in this fixture.
+    """This fixture returns the ids of all location relations, and if there is a defined distance.
+
+    The attribute id is indexed.
+
+    Specification details:
+
+    * If a user is assigned to a location and its children have related locations, their relations are included.
+    * If a user is assigned to a child location and its parent has related locations, the parent's relations are not included.
+    * This fixture will not contain any location specific data than the location's id
+    * Relations are two way
+
+    Example:
+    <fixture id="related_locations">
+      <locations>
+        <location id="location_a">
+          <related_location distance="X">location_b</related_location>
+        </location>
+        <location id="location_b">
+          <related_location distance="X">location_a</related_location>
+        </location>
+      </locations>
+    </fixture>
     """
     id = 'related_locations'
-    serializer = RelatedLocationSerializer()
 
     def __call__(self, restore_state):
         if not toggles.RELATED_LOCATIONS.enabled(restore_state.domain):
             return []
 
-        restore_user = restore_state.restore_user
-        user_locations = restore_user.get_sql_locations(restore_user.domain)
-        related_location_ids = LocationRelation.from_locations(user_locations)
-        related_location_pks = (
-            SQLLocation.objects.filter(location_id__in=related_location_ids)
-            .values_list('pk', flat=True)
+        location_relations = self._users_related_locations(restore_state.restore_user)
+
+        root_node = Element('fixture', {'id': self.id,
+                                        'user_id': restore_state.restore_user.user_id,
+                                        'indexed': 'true'})
+        outer_node = SubElement(root_node, 'locations')
+
+        for location, relations in location_relations:
+            location_node = SubElement(outer_node, 'location', {'id': location.location_id})
+            for location, distance in relations:
```

Comment:
nit: reusing the name `location` in this inner loop is confusing. The `location` in this loop is different from the `location` in the outer loop. Too bad stickler doesn't complain about that.

Reply:
oof yeah. that is a little confusing. will follow up